source/docs/hardware-reference/talonfx/improving-performance-with-current-limits.rst

```diff
@@ -0,0 +1,215 @@
+There exists 3 forms of current limiting: supply, stator and torque. For all limits, the following :ref:`configs <docs/tuner/configs:tuner configs>` apply. In the below example, replace ``Supply`` with ``Stator``.
+
+.. note:: Limits for torque based control modes should be applied with ``Peak Forward Torque Current`` and ``Peak Reverse Torque Current`` instead.
+
+- ``SupplyCurrentLimit``
+- ``SupplyCurrentLimitEnable``
+
+All limits must be enabled using the appropriate enable config.
+
+Supply Limits
+^^^^^^^^^^^^^
+
+.. note:: Supply limits are not functional in torque based control modes. Use the peak torque current configs instead.
+
+When ``SupplyCurrentThreshold`` has elapsed for ``SupplyTimeThreshold`` amount of time, it will lower motor output until it's within range of the current limit (see the above section on examples of these configs).
+
+Supply current limits are useful to prevent fuses from reseting and breakers from tripping. They are also effective at preventing brownouts and reducing overall current load on the battery.
+
+Stator & Torque Limits
+^^^^^^^^^^^^^^^^^^^^^^
+
+.. important::
+
+   Stator current limits are only applicable in non-torque control modes. For example, a stator limit would have no affect if the current control mode is ``TorqueCurrentFOC``. Users utilizing torque based control modes should use ``Peak Forward Torque Current`` and ``Peak Reverse Torque Current`` configs instead, but the following documentation is still applicable.
+
+Stator current is the output current (and proportional to torque) of the motor. The stator current differs from supply current in that it's at a different voltage from the input.
+
+.. note:: The below example ignores factors such as energy loss from heat or inefficiency.
+
+For example, if a motor is being applied with 50% dutycycle with :math:`80A`` of load (stator current), the resulting supply current will be :math:`40A`. If the motor has 100% dutycycle applied wth :math:`80A`` of stator current, the resulting supply current wll be :math:`80A`.
+
+.. figure:: images/stator-limit.png
+   :alt: 80a stator limit being applied
+
+   *80a torque limit taking affect*
+
+Stator limits apply the same limiting strategy used by supply limits, but are more effective in dealing with situations such as wheel slip or acceleration induced brownouts. The rationale behind it reducing acceleration induced brownouts is that initial acceleration is typically peak supply and stator current.
+
+If a robot acceleration event draws :math:`100A` of supply current, it will only do so for a very small portion of the entire acceleration event up to a maximum velocity. During the acceleration event, the motor is likely not being commanded with 100% dutycycle. Therefore, if :math:`100A` are being drawn at supply at :math:`6V`, then output stator is around :math:`200A`. A :math:`100A` stator limit would cap the supply current at `50A` while being more efficient at not limiting the entire robot velocity.
```

Comment:
I don't like the brownout mentions for stator current limit. It would be more valuable to demonstrate how stator current limits also have an impact on supply current draw. For example, an 80 A stator current limit results in ~15 A supply current at stall and 80 A at 4800 RPM, after which both currents drop rapidly.

Reply:
I focus pretty heavily on the assumption of 100% dutycycle means 1:1 stator/supply and 50% dutycycle means supply = stator / 2. 

I'm not sure how to phrase or refactor this paragraph to meet your suggestion.

Reply:
Supply current limits, at least at low velocity, actually have less of a restriction on acceleration than stator limits, and they are more effective at preventing brownouts. For example, a 50 A supply limit here could allow for more than 100 A stator at stall.

While you can explain the relationship between the two currents using duty cycle out, it's hard to do the same for current limits, because the current limits are what restrict duty cycle out. Stator limits do so linearly, while supply is nonlinear. Voltage is the dependent variable here, velocity and current limit are the independent variables.

Reply:
Oh lol I didn't actually mean for that to replace the text, that was more just me to you explaining why the original didn't feel right to me. I'll get something written up on Monday.